.github/workflows/ci.yml

```diff
@@ -0,0 +1,78 @@
+name: CI build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - support/*
+  workflow_dispatch:
+  workflow_call:
+
+env:
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+  DOTNET_CLI_TELEMETRY_OPTOUT: true
+  DOTNET_NOLOGO: true
+  DOTNET_GENERATE_ASPNET_CERTIFICATE: false
+
+jobs:
+
+  build-windows:
+    name: Build on Windows
+    runs-on: windows-2022
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+          fetch-depth: 0
+      - name: Run build script
+        run: ./build.cmd
+        shell: cmd
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: windows-artifacts
+          path: artifacts\**\*
+
+  build-ubuntu:
+    name: Build on Ubuntu
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+          fetch-depth: 0
+      - uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: |
+            5.0
+            3.1
+            2.1
+      - name: Run build script
+        run: ./build.sh
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: ubuntu-artifacts
+          path: artifacts/**/*
+
+  build-docs:
+    name: Build documentation
+    runs-on: ubuntu-22.04

```

Comment:
I see you prefer a named LTS version to "latest", eh?

Reply:
Yes, to avoid future issues when `ubuntu-latest` is updated to a newer version. Better to remain in control and update it when we want to, rather than having to fix things when they break because the image was updated behind our back.